Address Issue #492

tomjkidd opened the pull request:
This fixes issue #492, handling an invalid page parameter for both the html and json type of responses.

tobias commented on src/clojars/web.clj:
```diff
@@ -39,6 +39,21 @@
+    (let [i (Integer/parseInt maybe-page)]
+      i)
+    (catch NumberFormatException nfe
+      (let [e (ex-info
```
Is there a reason you used a `let` here (and above)? If not, it would be nice to get rid of those.

tomjkidd replied:
No real reason for above, for below, it was just to organize all the ex-info args while I was writing it. I can change it easily to remove them both.

tobias replied:
Great, thanks!

tobias commented:
We also return a 400 from the api here: https://github.com/clojars/clojars-web/tree/master/src/clojars/web/search.clj#L29. It would be nice to apply `error-api-response` there as well.

Would you be willing to add some tests for this as well?

tomjkidd commented:
I did see that api message, just wanted to get feedback before I changed too much.
As for tests, I have no problem trying to write some to cover the cases. I ran into issues using `lein test` though, because I am on Windows, so it'll be flying blind to some extent (notes from my experience in the issue)

tomjkidd commented:
I was able to figure out what was causing migrations to crash, so I also updated that to no be OS specific. I've also now added the tests. Thanks for the feedback.

tobias commented:
Merged, thanks!
